diff --git a/GPClient/CMakeLists.txt b/GPClient/CMakeLists.txt
index 578e1fd..052fdb9 100644
--- a/GPClient/CMakeLists.txt
+++ b/GPClient/CMakeLists.txt
@@ -46,6 +46,7 @@ add_3rdparty(
     SingleApplication
     GIT_REPOSITORY https://github.com/itay-grudev/SingleApplication.git
     GIT_TAG v3.3.0
+    BUILD_BYPRODUCT SingleApplication/src/SingleApplication-${PROJECT_NAME}-build/libSingleApplication.a
     CMAKE_ARGS
         -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
         -DCMAKE_CXX_FLAGS_RELEASE=${CMAKE_CXX_FLAGS_RELEASE}
diff --git a/GPService/CMakeLists.txt b/GPService/CMakeLists.txt
index 7f20655..aa656c3 100644
--- a/GPService/CMakeLists.txt
+++ b/GPService/CMakeLists.txt
@@ -38,6 +38,7 @@ add_3rdparty(
     SingleApplication
     GIT_REPOSITORY https://github.com/itay-grudev/SingleApplication.git
     GIT_TAG v3.3.0
+    BUILD_BYPRODUCT SingleApplication/src/SingleApplication-${PROJECT_NAME}-build/libSingleApplication.a
     CMAKE_ARGS
         -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
         -DCMAKE_CXX_FLAGS_RELEASE=${CMAKE_CXX_FLAGS_RELEASE}
diff --git a/cmake/Add3rdParty.cmake b/cmake/Add3rdParty.cmake
index 3809a14..2d3f898 100644
--- a/cmake/Add3rdParty.cmake
+++ b/cmake/Add3rdParty.cmake
@@ -1,7 +1,7 @@
 include(ExternalProject)
 
 function(add_3rdparty NAME)
-    set(oneValueArgs GIT_REPOSITORY GIT_TAG)
+    set(oneValueArgs GIT_REPOSITORY GIT_TAG BUILD_BYPRODUCT)
     cmake_parse_arguments(add_3rdparty_args "" "${oneValueArgs}" "" ${ARGN})
 
     if(EXISTS "${CMAKE_SOURCE_DIR}/3rdparty/${NAME}/CMakeLists.txt")
@@ -11,6 +11,7 @@ function(add_3rdparty NAME)
             ${NAME}-${PROJECT_NAME}
             PREFIX ${CMAKE_CURRENT_BINARY_DIR}/${NAME}
             SOURCE_DIR "${CMAKE_SOURCE_DIR}/3rdparty/${NAME}"
+            BUILD_BYPRODUCTS ${add_3rdparty_args_BUILD_BYPRODUCT}
             INSTALL_COMMAND ""
             "${add_3rdparty_args_UNPARSED_ARGUMENTS}"
         )
